### PR TITLE
Cy articles create db

### LINKED ISSUE #38: Create database table for Articles
# Dependencies

This issue should be completed *after* the issue "Articles - Create personal dokku dev deployment"

# Overview

This issue involves creating the database table and migration file for `Articles`.

For step-by-step instructions, please see these portions of the
lab writeup:

* <https://ucsb-cs156.github.io/f24/lab/team01.html#the-two-database-tables-in-the-starting-code>
* <https://ucsb-cs156.github.io/f24/lab/team01.html#your-task-add-crud-for-additional-database-tables>
* <https://ucsb-cs156.github.io/f24/lab/team01.html#adding-a-database-table>

# Acceptance Criteria:

- [x] There is an `@Entity` class called Articles.java
- [x] There is a `@Repository` class called ArticlesRepository.java
- [x] There is a database migration file called Articles.json
- [x] When you start up the repo on localhost, you can see the table
      using the H2 console (see the file [`docs/h2-database.md`](/ucsb-cs156-f24/STARTER-team01//blob/main/docs/h2-database.md) for 
      instructions.)
- [x] When you deploy the app to Dokku, you can see the table
      using the `dokku postgres:connect team01-db` command
      (use `\dt` to list the tables) (See <https://ucsb-cs156.github.io/topics/dokku/postgres_database.html#postgres-command-line> for more info.)

# What to do next

When you've met all of the acceptance criteria:

**Make a PR**: Do a PR for this branch, being sure to enter a [reasonable title](https://ucsb-cs156.github.io/topics/pull_requests.html#pr-titles)
and [description](https://ucsb-cs156.github.io/topics/pull_requests.html#pr-descriptions).

In the description, include the text "Closes #n" where `n` is the number of
this issue.

Then ask for a team member to review your PR.

Finally: drag this issue from "In Progress" to "In Review".

**Review Others' PRs**: Next, see if there are any PRs that need to be code reviewed and merged.  Unless you are the
first person on your team to complete a PR, there likely is at least one.
See: 
* [General Code Review Advice](https://ucsb-cs156.github.io/topics/code_reviews/)
* [Performin Code Reviews on Github](https://ucsb-cs156.github.io/topics/code_reviews/code_reviews_github.html)

**Start your next issue **: Then, find the issue for 
"Create `ArticlesController..." on the Kanban board in the ToDo column.

* Assign it to yourself
* Drag into "In Progress"
* Create a new branch, starting from the branch for the issue you just finished.
* Start on the new issue.

### PR DESCRIPTION
**Pull request by qwistaycat**
Closes #38

### CODE REVIEW

**Review by caleb-stahl — COMMENTED** [review]
Same questions as your other PR. I think it might need to be named ARTICLES_PK but maybe double check to be sure. Other stuff looks good.

File: src/main/java/edu/ucsb/cs156/example/controllers/ArticlesController.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+import jakarta.validation.Valid;
+
+import java.time.LocalDateTime;
+
+@Tag(name = "Articles")
+@RequestMapping("/api/articles")
+@RestController
+@Slf4j
+public class ArticlesController {
```
Comment (obenedek20) [inline src/main/java/edu/ucsb/cs156/example/controllers/ArticlesController.java]: ArticlesController should extend ApiController

File: src/main/java/edu/ucsb/cs156/example/repositories/UCSBDateRepository.java [hunk 2]
```diff
@@ -5,16 +5,7 @@
...
- * The UCSBDateRepository is a repository for UCSBDate entities.
- */
-
 @Repository
 public interface UCSBDateRepository extends CrudRepository<UCSBDate, Long> {
-  /**
-   * This method returns all UCSBDate entities with a given quarterYYYYQ.
-   * @param quarterYYYYQ quarter in the format YYYYQ (e.g. 20241 for Winter 2024, 20242 for Spring 2024, 20243 for Summer 2024, 20244 for Fall 2024)
-   * @return all UCSBDate entities with a given quarterYYYYQ
-   */
```
Comment (obenedek20) [inline src/main/java/edu/ucsb/cs156/example/repositories/UCSBDateRepository.java]: did you mean to remove these?

File: src/main/resources/db/migration/changes/Articles.json [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+          "changes": [
+            {
+              "createTable": {
+                "columns": [
+                  {
+                    "column": {
+                      "autoIncrement": true,
+                      "constraints": {
+                        "primaryKey": true,
+                        "primaryKeyName": "CONSTRAINT_6"
```
Comment (obenedek20) [inline src/main/resources/db/migration/changes/Articles.json]: I think he updated the instructions for naming Primary Keys so this should be ARTICLES_PK

File: src/main/resources/db/migration/changes/UCSBDates.json [hunk 4]
```diff
@@ -27,7 +27,7 @@
                       "autoIncrement": true,
                       "constraints": {
                         "primaryKey": true,
-                        "primaryKeyName": "CONSTRAINT_5"
+                        "primaryKeyName": "CONSTRAINT_7"
```
Comment (obenedek20) [inline src/main/resources/db/migration/changes/UCSBDates.json]: Might need to pull from main, this is a fix they implemented a few days ago so shouldn't change it

Could also just unstage it

**Review by obenedek20 — CHANGES_REQUESTED** [review]

**Review by obenedek20 — APPROVED** [review]

**Pull request merged**